backend/restApi.js

```diff
@@ -0,0 +1,116 @@
+  const gameIdElements = ["", "", ""].map((x) => {
+    const wordIndex = Math.floor(WORDS.length * Math.random());
+    return WORDS[wordIndex];
+  });
+  const gameId = gameIdElements.join("-");
+  return gameId;
+};
+
+router.get("/game/:gameId", (req, res) => {
+  const { gameId } = req.params;
+  getGame(gameId, (err, rows) => {
+    if (err || rows.length === 0) {
+      console.error(`GET /game: Unable to find game ${gameId}`);
+      res.status = 400;
+      res.send("Unable to find game");
+      return;
+    }
+    const crosswordId = rows[0].crossword;
+    getCrossword(crosswordId, (rows) => {
+      const crossword = rows[0];
+      const response = {
+        gameId,
+        crossword
+      };
+      res.contentType = "appliction/json";
+      res.send(JSON.stringify(response));
+    });
+  });
+});
+
+router.post("/game", (req, res) => {
+  const { crosswordId } = req.body;
+  const gameId = generateGameId();
```

Comment:
Retry if a duplicate id is created? Highly unlikely so a bit of an premature optimization but...

Reply:
Keeping this as is for now